### PR TITLE
📦 NEW: Batch sync mechanism

### PR DESCRIPTION
**Pull request by irshadCypherock**

### CODE REVIEW

**Comment by irshadCypherock** [comment]
Task linked: [CSI-982 Implement batch sync mechanism in cysync](https://app.clickup.com/t/37308523/CSI-982)

File: cysync/src/store/provider/syncProvider/types/syncItem.ts [hunk 1]
```diff
@@ -23,4 +23,12 @@ export abstract class SyncItem {
     this.module = module;
     this.retries = 0;
   }
+
+  equals(_item: SyncItem) {
+    throw new Error('Please override this function');
```
Comment (Suraj-Tiwari) [inline cysync/src/store/provider/syncProvider/types/syncItem.ts:28]: equals not implemented for this class

File: cysync/src/store/provider/syncProvider/types/syncItem.ts [hunk 2]
```diff
@@ -23,4 +23,12 @@ export abstract class SyncItem {
...
+  }
+
+  clone() {
+    throw new Error('Please override this function');
```
Comment (Suraj-Tiwari) [inline cysync/src/store/provider/syncProvider/types/syncItem.ts:32]: clone not implemented for this class

**Review by Suraj-Tiwari — APPROVED** [review]
merge it after updating message

**Pull request merged**